contrib/experimental/great_expectations_experimental/expectations/expect_queried_pair_column_values_to_have_diff.py

```diff
@@ -0,0 +1,142 @@
+        "diff": 0,
+        "mostly": 1,
+        "strict": True,
+        "query": query,
+    }
+
+    def validate_configuration(
+        self, configuration: Optional[ExpectationConfiguration]
+    ) -> None:
+        super().validate_configuration(configuration)
+        diff = configuration["kwargs"].get("diff")
+        mostly = configuration["kwargs"].get("mostly")
+
+        try:
+            assert diff is not None, "'diff' must be specified"
+            assert isinstance(diff, (int, float)), "`diff` must be a valid float or int"
+            assert (
+                isinstance(mostly, (int, float)) and 0 < mostly <= 1
+            ), "'mostly' must be 1, a float between 0 and 1"
+        except AssertionError as e:
+            raise InvalidExpectationConfigurationError(str(e))
+
+    def _validate(
+        self,
+        configuration: ExpectationConfiguration,
+        metrics: dict,
+        runtime_configuration: dict = None,
+        execution_engine: ExecutionEngine = None,
+    ) -> Union[ExpectationValidationResult, dict]:
+
+        diff = configuration["kwargs"].get("diff")
+        mostly = configuration["kwargs"].get("mostly")
+        strict = configuration["kwargs"].get("strict")
+        query_result = metrics.get("query.pair_column")
+
+        if mostly == 1:
+            success = all([(abs(x[0]) == diff) for x in query_result])
+        elif mostly != 1 and strict is True:
+            success = (
+                sum([(abs(x[0]) == diff) for x in query_result]) / len(query_result)
+            ) > mostly
+        else:
+            success = (
+                sum([(abs(x[0]) == diff) for x in query_result]) / len(query_result)
+            ) >= mostly
+
+        return {
+            "success": success,
+            "result": {"observed_value": [x[0] for x in query_result]},
+        }
+
+    examples = [
+        {
+            "data": [
+                {
+                    "dataset_name": "test",
+                    "data": {
+                        "col1": [1, 2, 2, 3, 4],
+                        "col2": [2, 3, 3, 4, 2],
+                    },
+                },
+            ],
+            "tests": [
+                {
+                    "title": "basic_positive_test",
+                    "exact_match_out": False,
+                    "include_in_gallery": True,
+                    "in": {
+                        "column_A": "col1",
+                        "column_B": "col2",
+                        "diff": 1,
+                        "mostly": 0.6,
+                    },
+                    "out": {"success": True},
+                    "only_for": ["sqlite", "spark"],
+                },
+                {
+                    "title": "basic_negative_test",
+                    "exact_match_out": False,
+                    "include_in_gallery": True,
+                    "in": {
+                        "column_A": "col1",
+                        "column_B": "col2",
```

Comment:
@austiezr Should we add some tests that are floating point numbers?  Are there checks against non-applicable types in the column?  Thanks.